### ISSUE TITLE
Error with Linked Entity Retrieval when the relationship is a list and using pick on the relationship attribute

### ISSUE THREAD

**Issue by alifrugal** [ScorpioBroker/ScorpioBroker#566]
Using the Linked Entity Retrieval with an array of relationship raises an internal error. 

```
{
    "type": "https://uri.etsi.org/ngsi-ld/errors/InternalError",
    "title": "Internal error",
    "detail": null,
    "status": 500
}
```

How to reproduce: 

Create a entity with a list of relationships, in the example below I create a building with a relationship `hasRooms` that contains an array of three rooms. 

```
{
  "id": "urn:ngsi-ld:Building:Office-1",
  "type": "Building",
  "alternateName": {
    "value": "Office 1"
  },
  "hasRooms": {
    "object": [
        "urn:ngsi-ld:Room:Office-1-Room-001",
        "urn:ngsi-ld:Room:Office-1-Room-002",
        "urn:ngsi-ld:Room:Office-1-Room-003"
    ],
    "objectType":"Room"
  }
}
```

Then create the three rooms, example of Room-1: 
```
    {
  "id": "urn:ngsi-ld:Room:Office-1-Room-001",
  "type": ["Room", "MeetingRoom"],
  "alternateName": {
      "value": "Room-001"
  },
  "isPartOf": { 
    "object": ["urn:ngsi-ld:Building:Office-1"],
    "objectType": "Building"
  }
}
```

When retrieval the building entity and retrieve its linked entities picking the `hasRooms` as the attributes to return, it raises an internal error with the query: `ngsi-ld/v1/entities/urn:ngsi-ld:Building:Office-1?join=inline&joinLevel=2&pick=id,type,hasRooms`

And also with selecting the attribute within the linked entity: 
`
/ngsi-ld/v1/entities/urn:ngsi-ld:Building:Office-1?join=inline&joinLevel=2&pick=id,hasRooms%7Bid%7D
`
with the error: 
{
    "type": "https://uri.etsi.org/ngsi-ld/errors/InternalError",
    "title": "Internal error",
    "detail": "eu.neclab.ngsildbroker.commons.exceptions.ResponseException",
    "status": 500
}

Hint: I think that the first error happens in the `QueryService` class in `getEntityInline` function. 
The line `attrsMap.remove(toRemove);` triggers the error. 
the second one seems to happen at line: 
`unisOfMaps.add(getEntityInline(context, tenant, (String) idsMap.get(JsonLdConsts.VALUE), finalAttrsMap, lang, localOnly,
containedBy + ent.get(JsonLdConsts.ID), joinLvl.get()).onItem()
`

**Comment by brijeshnec** [comment]
Hi,

The issue has been resolved in PR #568 
Please update your GET request to the following format:
/ngsi-ld/v1/entities/urn:ngsi-ld:Building:Office-1?join=inline&joinLevel=2&pick=id,hasRooms%7B**entity.id**%7D 

Thank you.

**Comment by alifrugal** [comment]
Thanks @brijeshnec, another issue is that if attributes of the main entity come after the ones of the linked entity, then they are ignored. 

For example: 
`/ngsi-ld/v1/entities/urn:ngsi-ld:Building:Office-1?join=inline&joinLevel=2&pick=id,hasRooms%7Bentity.id%7D,alternateName` would not give the attribute `alternateName` of the entity `urn:ngsi-ld:Building:Office-1` while the query `/ngsi-ld/v1/entities/urn:ngsi-ld:Building:Office-1?join=inline&joinLevel=2&pick=id,alternateName,hasRooms%7Bentity.id%7D` would  give it back
